cirq-google/cirq_google/engine/engine_sampler.py

```diff
@@ -139,3 +139,106 @@ def get_engine_sampler(
         )
     gate_set = gate_sets.NAMED_GATESETS[gate_set_name]
     return engine.get_engine(project_id).sampler(processor_id=processor_id, gate_set=gate_set)
+
+
+def get_device_sampler(
+    project_id: Optional[str] = None, processor_id: Optional[str] = None
+) -> Tuple[
+    Tuple[Union[cirq.Device], int],
+    Union['cirq_google.PhasedFSimEngineSimulator', 'cirq_google.QuantumEngineSampler'],
+    bool,
```

Comment:
Returning a complicated tuple like this seems less than ideal. I'd suggest defining a dataclass to return instead:
```python
@dataclasses.dataclass
class DeviceSamplerInfo:
    device: cirq.Device
    device_line_length: int
    sampler: Union[cirq_google.PhasedFSimEngineSimulator, cirq_google.QuantumEngineSampler]
    signed_in: bool

def get_device_sampler(...) -> DeviceSamplerInfo:
    ...
```
Then when you use this you can do things like:
```python
result = get_device_sampler(...)
if not result.signed_in:
    ...
```
This makes it easier to evolve the API in the future, e.g. if we decide we need to return more info we can add fields to the dataclass without needing to change any callsites.